commands/project/upload.ts

```diff
@@ -1,64 +1,70 @@
+import {
   ensureProjectExists,
   getProjectConfig,
   logFeedbackMessage,
   validateProjectConfig,
-} = require('../../lib/projects');
-const { handleProjectUpload } = require('../../lib/projects/upload');
-const {
+} from '../../lib/projects';
+import { handleProjectUpload } from '../../lib/projects/upload';
+import {
   displayWarnLogs,
   pollProjectBuildAndDeploy,
-} = require('../../lib/projects/buildAndDeploy');
-const { i18n } = require('../../lib/lang');
-const { getAccountConfig } = require('@hubspot/local-dev-lib/config');
-const { isSpecifiedError } = require('@hubspot/local-dev-lib/errors/index');
-const { PROJECT_ERROR_TYPES } = require('../../lib/constants');
-const { logError, ApiErrorContext } = require('../../lib/errorHandlers/index');
-const { EXIT_CODES } = require('../../lib/enums/exitCodes');
-
+} from '../../lib/projects/buildAndDeploy';
+import { i18n } from '../../lib/lang';
+import { PROJECT_ERROR_TYPES } from '../../lib/constants';
+import { logError, ApiErrorContext } from '../../lib/errorHandlers/index';
+import { EXIT_CODES } from '../../lib/enums/exitCodes';
+import { CommonArgs } from '../../types/Yargs';
+import { ProjectPollResult } from '../../types/Projects';
+import { makeYargsBuilder } from '../../lib/yargsUtils';
 const i18nKey = 'commands.project.subcommands.upload';
 
-exports.command = 'upload';
-exports.describe = uiBetaTag(i18n(`${i18nKey}.describe`), false);
+export const command = 'upload';
+export const describe = uiBetaTag(i18n(`${i18nKey}.describe`), false);
 
-exports.handler = async options => {
-  const { forceCreate, message, derivedAccountId, skipValidation } = options;
+type ProjectUploadArgs = CommonArgs & {
+  forceCreate: boolean;
+  message: string;
+  m: string;
+  skipValidation: boolean;
+};
+
+export async function handler(
+  args: ArgumentsCamelCase<ProjectUploadArgs>
+): Promise<void> {
+  const { forceCreate, message, derivedAccountId, skipValidation } = args;
   const accountConfig = getAccountConfig(derivedAccountId);
   const accountType = accountConfig && accountConfig.accountType;
 
   const { projectConfig, projectDir } = await getProjectConfig();
 
-  trackCommandUsage('project-upload', { type: accountType }, derivedAccountId);
+  trackCommandUsage('project-upload', { type: accountType! }, derivedAccountId);
 
   validateProjectConfig(projectConfig, projectDir);
 
-  await ensureProjectExists(derivedAccountId, projectConfig.name, {
+  await ensureProjectExists(derivedAccountId, projectConfig!.name, {
     forceCreate,
     uploadCommand: true,
   });
 
   try {
-    const { result, uploadError } = await handleProjectUpload(
-      derivedAccountId,
-      projectConfig,
-      projectDir,
-      pollProjectBuildAndDeploy,
-      message,
-      useV3Api(projectConfig?.platformVersion),
-      skipValidation
-    );
+    const { result, uploadError } =
+      await handleProjectUpload<ProjectPollResult>(
+        derivedAccountId,
+        projectConfig!,
```

Comment:
It's not the end of the world (and doesn't need to be fixed in this PR), but I really think we should be try to minimize our use of the `!` operator and instead write code where TS can infer these things exists. In this instance, we know `projectConfig` exists, but we could rewrite `valildateProjectConfig` in such a way that TS knows too. There are other instances where assert non-null when we _don't_ know - I think in those, manually checking is better, even if it's more verbose

Reply:
@camden11 yeah makes sense. I've been mostly trying to make these changes without having any impact on the command behavior. I realized that we can get an easy win just by adding an assertion to the return type of validateProjectConfig though! I added that in [4f24372](https://github.com/HubSpot/hubspot-cli/pull/1416/commits/4f243722e75e60d1edb73c3f2a2abebda0ab5b43)

I couldn't get it to also assert projectDir unfortunately, but this is still better. From what I can tell, the only options are to either split this out into two functions that each assert one of the inputs, or to change the input args to an object that contains both inputs.